src/overlays/actors/ovl_En_Bjt/z_en_bjt.c

```diff
@@ -31,49 +58,408 @@ const ActorInit En_Bjt_InitVars = {
+        case TOILET_HAND_BEHAVIOUR_WAIT_FOR_ITEM:
+            switch (Message_GetState(&play->msgCtx)) {
+                case 4:
+                case 5:
+                    if (!Message_ShouldAdvance(play)) {
+                        break;
+                    }
+                case 16:
+                    itemAP = func_80123810(play);
+                    if ((itemAP == EXCH_ITEM_DEED_LAND) || (itemAP == EXCH_ITEM_LETTER_TO_KAFEI) ||
+                        (itemAP == EXCH_ITEM_DEED_SWAMP) || (itemAP == EXCH_ITEM_DEED_MOUNTAIN) ||
+                        (itemAP == EXCH_ITEM_DEED_OCEAN) || (itemAP == EXCH_ITEM_LETTER_MAMA)) {
+                        EnBjt_ChangeAnimation(this, TOILET_HAND_ANIM_WAITING_MORPH);
+                        this->playedSfx = false;
+                        this->behaviour++;
+                        scriptBranch = 1; // Right item
+                    } else if (itemAP < 0) {
+                        this->playedSfx = false;
+                        this->behaviour++;
+                        scriptBranch = 3; // Not showing item
+                    } else if (itemAP != 0) {
+                        this->playedSfx = false;
+                        this->behaviour++;
+                        scriptBranch = 2; // Wrong item
+                    }
+                    break;
+
+                default:
+                    break;
+            }
+            break;
+
+        case TOILET_HAND_BEHAVIOUR_TAKE_ITEM:
+            if (player->exchangeItemId != EXCH_ITEM_NONE) {
+                EnBjt_TakeItem(player->exchangeItemId);
+                player->exchangeItemId = EXCH_ITEM_NONE;
+            }
+            if (EnBjt_Vanish(this)) {
+                this->timer = 60;
+                this->behaviour++;
+                scriptBranch = 1; // Right item
+            }
+            break;
+
+        case TOILET_HAND_BEHAVIOUR_USE_ITEM:
+            if (DECR(this->timer) == 0) {
+                EnBjt_ChangeAnimation(this, TOILET_HAND_ANIM_FIST); // change while not visible
+                this->playedSfx = false;
+                this->behaviour++;
+            } else if (this->timer == 10) {
+                Actor_PlaySfxAtPos(&this->actor, NA_SE_EV_TOILET_WATER);
+            }
+            break;
+
+        case TOILET_HAND_BEHAVIOUR_REAPPEAR:
+            if (EnBjt_Appear(this)) {
+                this->behaviour++;
+                scriptBranch = 1; // Right item
+            }
+            break;
+
+        case TOILET_HAND_BEHAVIOUR_REWARD_GIVEN:
+            EnBjt_ChangeAnimation(this, TOILET_HAND_ANIM_OPEN_HAND);
+            this->behaviour++;
+            scriptBranch = 1; // Right item
+            break;
+
+        default:
+            break;
+    }
+
+    return scriptBranch;
+}
+
+// Used in Update
+s32 EnBjt_CheckTalk(EnBjt* this, PlayState* play) {
+    s32 ret = false;
+
+    if ((this->stateFlags & 7) && Actor_ProcessTalkRequest(&this->actor, &play->state)) {
```

Comment:
Should you have a define for `7` (stateFlags)?

Reply:
As I said twice above, this seems to be a SubS-general thing, not related to this actor, so I've left it for docs of that.

Reply:
After this goes in I might look into making an enum for this SubS stuff. It will just be placeholders since that SubS function is one of the few undocumented as it needs docs from other places.